numpyro/distributions/constraints.py

```diff
@@ -67,9 +83,12 @@ def __call__(self, x):
         return np.all(x > 0, axis=-1) & (x_sum <= 1) & (x_sum > 1 - 1e-6)
 
 
+boolean = _Boolean()
```

Comment:
Nit: we can make this _IntegerInterval(0, 1)

Reply:
The issue is that it will be much slower, almost 5X or so.